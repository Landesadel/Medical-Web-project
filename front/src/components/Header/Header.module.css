.header {
  background: rgba(255, 255, 255, 0.9);
  background: -webkit-gradient(linear, left top, right bottom, from(white), to(rgba(209, 200, 200, 0.9)));
  background: linear-gradient(to bottom right, white, rgba(209, 200, 200, 0.9));
  text-align: center;
  padding: 6px 0;
  -webkit-box-shadow: inset 0 0 10px white;
          box-shadow: inset 0 0 10px white;
  position: -webkit-sticky;
  position: sticky;
  z-index: 5;
  top: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.header form {
  position: fixed;
  top: 0;
  width: 100%;
  background-color: #ffffff;
  border-bottom: 1px solid #9ea6e4;
}

.logo {
  width: 130px;
  -o-object-fit: contain;
     object-fit: contain;
}

.headerSearchBtn,
.headerSearchBtnHandler {
  width: 60px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  border: none;
  outline: none;
}

.headerSearchBtnHandler {
  margin: 0 auto;
}

.headerSearchInput {
  border: none;
  outline: none;
  width: 100%;
}

.headerAccountIcon {
  width: 32px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

a {
  color: rgba(51, 51, 51, 0.9);
  font-size: 22px;
  font-weight: 500;
}

a:hover {
  color: rgba(68, 68, 167, 0.5);
}

ul {
  gap: 36px;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

ul div {
  font-size: 22px;
  font-weight: 500;
  color: rgba(51, 51, 51, 0.9);
}

ul div:hover {
  color: rgba(68, 68, 167, 0.5);
}

svg {
  fill: rgba(51, 51, 51, 0.9);
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

svg:hover {
  -webkit-transform: scale(1.2);
          transform: scale(1.2);
  fill: rgba(68, 68, 167, 0.5);
}
/*# sourceMappingURL=Header.module.css.map */