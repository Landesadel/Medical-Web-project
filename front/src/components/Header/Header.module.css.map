{
    "version": 3,
    "mappings": "AAOA,AAAA,OAAO,CAAC;EANN,UAAU,EAAE,wBAAwB;EACpC,UAAU,EAAE,kEAAqF;EACjG,UAAU,EAAE,+DAAkF;EAC9F,UAAU,EAAE,iEAAsF;EAKlG,UAAU,EAAE,MAAM;EAClB,OAAO,EAAE,KAAK;EACd,UAAU,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,CAAC,IAAI,CAAC,KAAsB;EACjD,QAAQ,EAAE,MAAM;EAChB,OAAO,EAAE,CAAC;EACV,GAAG,EAAE,CAAC;EACN,UAAU,EAAE,aAAa;CAS1B;;AAjBD,AAUE,OAVK,CAUL,IAAI,CAAC;EACH,QAAQ,EAAE,KAAK;EACf,GAAG,EAAE,CAAC;EACN,KAAK,EAAE,IAAI;EACX,gBAAgB,EAAE,OAAO;EACzB,aAAa,EAAE,GAAG,CAAC,KAAK,CAAC,OAAsB;CAChD;;AAGH,AAAA,KAAK,CAAC;EACJ,KAAK,EAAE,KAAK;EACZ,UAAU,EAAE,OAAO;CACpB;;AAED,AAAA,gBAAgB;AAChB,uBAAuB,CAAC;EACtB,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;EACvB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;CACd;;AAED,AAAA,uBAAuB,CAAC;EACtB,MAAM,EAAE,MAAM;CACf;;AAED,AAAA,kBAAkB,CAAC;EACjB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,KAAK,EAAE,IAAI;CACZ;;AAED,AAAA,kBAAkB,CAAC;EACjB,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;CACxB;;AAED,AAAA,CAAC,CAAC;EACA,KAAK,EAAE,qBAAqB;EAC5B,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,GAAG;CAKjB;;AARD,AAKE,CALD,AAKE,MAAM,CAAC;EACN,KAAK,EAAE,sBAAsB;CAC9B;;AAGH,AAAA,EAAE,CAAC;EACD,GAAG,EAAE,IAAI;EACT,WAAW,EAAE,MAAM;CAWpB;;AAbD,AAIE,EAJA,CAIA,GAAG,CAAC;EACF,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,GAAG;EAChB,KAAK,EAAE,qBAAqB;CAK7B;;AAZH,AASI,EATF,CAIA,GAAG,AAKA,MAAM,CAAC;EACN,KAAK,EAAE,sBAAsB;CAC9B;;AAKL,AAAA,GAAG,CAAC;EACF,IAAI,EAAE,qBAAqB;EAC3B,UAAU,EAAE,IAAI;CAKjB;;AAPD,AAGE,GAHC,AAGA,MAAM,CAAC;EACN,SAAS,EAAE,UAAU;EACrB,IAAI,EAAE,sBAAsB;CAC7B",
    "sources": [
        "Header.module.scss"
    ],
    "names": [],
    "file": "Header.module.css"
}