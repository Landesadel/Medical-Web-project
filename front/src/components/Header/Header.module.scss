@mixin headerBackgroundGradient() {
  background: rgba(255, 255, 255, 1.0);
  background: -webkit-linear-gradient(top left, rgba(255, 255, 255, 1.0), rgba(209, 200, 200, 1.0));
  background: -moz-linear-gradient(top left, rgba(255, 255, 255, 1.0), rgba(209, 200, 200, 1.0));
  background: linear-gradient(to bottom right, rgba(255, 255, 255, 1.0), rgba(209, 200, 200, 1.0));
}

.header {
  @include headerBackgroundGradient;
  text-align: center;
  padding: 12px 0;
  position: sticky;
  top: 0;
  form{
    position: fixed;
    top: 0;
    width: 100%;
    background-color: #ffffff;
    border-bottom: 1px solid rgba(158, 166, 228, 1);
  }
}

.logo {
  width: 130px;
  object-fit: contain;
}

.headerSearchBtn, .headerSearchBtnHandler {
  width: 60px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  outline: none;
}

.headerSearchBtnHandler {
  margin: 0 auto;
}

.headerSearchInput {
  border: none;
  outline: none;
  width: 100%;
}

.headerAccountIcon {
  width: 32px;
  display: flex;
  align-items: center;
  justify-content: center;
}

