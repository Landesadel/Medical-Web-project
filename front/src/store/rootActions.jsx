import * as userActions from './user/user.actions';
import * as forumActions from './forum/forum.actions';

export const allActions = {
	...userActions,
	...forumActions,
};
