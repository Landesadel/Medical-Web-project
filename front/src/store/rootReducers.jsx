import { reducer as userReducer } from './user/user.slice';
import { reducer as forumReducer } from './forum/forum.slice';
import { reducer as toastrReducer } from 'react-redux-toastr';

export const reducers = {
	user: userReducer,
	toastr: toastrReducer,
	forum: forumReducer,
};
