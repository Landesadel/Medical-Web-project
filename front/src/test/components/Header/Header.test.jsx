import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {BrowserRouter} from 'react-router-dom';
import Header from "../../../components/Header/Header";
import {Provider} from "react-redux";
import { store } from '../../../store/store.jsx';



describe('Header', () => {
    it('render header', () => {
       const snapshot= render(
            <Provider store={store}>
            <BrowserRouter>
                <Header/>
            </BrowserRouter>
            </Provider>
        );

        expect(screen.getByText('Главная')).toBeInTheDocument();
        expect(snapshot).toMatchSnapshot();
    });


});
