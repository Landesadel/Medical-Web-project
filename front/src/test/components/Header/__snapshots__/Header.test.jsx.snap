// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
        id="header"
      >
        <nav
          class="navbar navbar-expand-lg h-100 navbar-custom"
        >
          <div
            class="container p-lg-0"
          >
            <a
              class="navbar-brand logo"
              href="/"
            >
              <img
                alt="Bosom"
                src=""
              />
            </a>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse d-lg-flex justify-content-lg-between"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto ms-lg-2 mb-2 mb-lg-0 w-100 justify-content-center"
              >
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="menu_link"
                    href="/"
                  >
                    Главная
                  </a>
                </li>
                <li
                  class="nav-item dropdown"
                >
                  <div
                    aria-expanded="false"
                    class="menu_link dropdown-toggle"
                    data-bs-toggle="dropdown"
                    role="button"
                  >
                    Общество
                  </div>
                  <ul
                    class="dropdown-menu dropdown-menu-right"
                  >
                    <li>
                      <a
                        class="dropdown-item"
                        href="/statute"
                      >
                        Устав
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/structure"
                      >
                        Структура
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/normatives"
                      >
                        Нормативные документы
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/partners"
                      >
                        Компании-партнеры
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/register"
                      >
                        Вступить в общество
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/payments"
                      >
                        Оплатить взносы
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/photos"
                      >
                        Фотогалерея
                      </a>
                    </li>
                  </ul>
                </li>
                <li
                  class="nav-item dropdown"
                >
                  <div
                    aria-expanded="false"
                    class="menu_link dropdown-toggle"
                    data-bs-toggle="dropdown"
                    role="button"
                  >
                    Обучение
                  </div>
                  <ul
                    class="dropdown-menu dropdown-menu-right"
                  >
                    <li>
                      <a
                        class="dropdown-item"
                        href="/centers"
                      >
                        Симуляционные центры
                      </a>
                    </li>
                    <li>
                      <a
                        class="dropdown-item"
                        href="/study"
                      >
                        Учебные материалы
                      </a>
                    </li>
                  </ul>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="menu_link"
                    href="/conferencies"
                  >
                    Конференции
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="menu_link"
                    href="/forum"
                  >
                    Форум
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="menu_link"
                    href="/contacts"
                  >
                    Контакты
                  </a>
                </li>
              </ul>
              <div
                class="nav-item ms-auto headerAccountIcon"
              >
                <a
                  aria-current="page"
                  class="nav-link"
                  href="/login"
                >
                  <svg
                    height="32px"
                    version="1.1"
                    viewBox="0 0 1024 1024"
                    width="32px"
                  >
                    <path
                      d="M170.666667 512a42.666667 42.666667 0 0 0 42.666666 42.666667h323.84l-98.133333 97.706666a42.666667 42.666667 0 0 0 0 60.586667 42.666667 42.666667 0 0 0 60.586667 0l170.666666-170.666667a42.666667 42.666667 0 0 0 8.96-14.08 42.666667 42.666667 0 0 0 0-32.426666 42.666667 42.666667 0 0 0-8.96-14.08l-170.666666-170.666667a42.666667 42.666667 0 1 0-60.586667 60.586667l98.133333 97.706666H213.333333a42.666667 42.666667 0 0 0-42.666666 42.666667zM725.333333 85.333333H298.666667a128 128 0 0 0-128 128v128a42.666667 42.666667 0 0 0 85.333333 0V213.333333a42.666667 42.666667 0 0 1 42.666667-42.666666h426.666666a42.666667 42.666667 0 0 1 42.666667 42.666666v597.333334a42.666667 42.666667 0 0 1-42.666667 42.666666H298.666667a42.666667 42.666667 0 0 1-42.666667-42.666666v-128a42.666667 42.666667 0 0 0-85.333333 0v128a128 128 0 0 0 128 128h426.666666a128 128 0 0 0 128-128V213.333333a128 128 0 0 0-128-128z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
          <form
            class="d-none"
            role="search"
          >
            <input
              aria-label="Search"
              class="me-2 headerSearchInput"
              placeholder="Поиск"
              type="search"
            />
            <button
              class="btn headerSearchBtn"
              type="submit"
            >
              <img
                alt="search"
                src=""
              />
            </button>
          </form>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
      id="header"
    >
      <nav
        class="navbar navbar-expand-lg h-100 navbar-custom"
      >
        <div
          class="container p-lg-0"
        >
          <a
            class="navbar-brand logo"
            href="/"
          >
            <img
              alt="Bosom"
              src=""
            />
          </a>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarSupportedContent"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse d-lg-flex justify-content-lg-between"
            id="navbarSupportedContent"
          >
            <ul
              class="navbar-nav me-auto ms-lg-2 mb-2 mb-lg-0 w-100 justify-content-center"
            >
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="menu_link"
                  href="/"
                >
                  Главная
                </a>
              </li>
              <li
                class="nav-item dropdown"
              >
                <div
                  aria-expanded="false"
                  class="menu_link dropdown-toggle"
                  data-bs-toggle="dropdown"
                  role="button"
                >
                  Общество
                </div>
                <ul
                  class="dropdown-menu dropdown-menu-right"
                >
                  <li>
                    <a
                      class="dropdown-item"
                      href="/statute"
                    >
                      Устав
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/structure"
                    >
                      Структура
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/normatives"
                    >
                      Нормативные документы
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/partners"
                    >
                      Компании-партнеры
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/register"
                    >
                      Вступить в общество
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/payments"
                    >
                      Оплатить взносы
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/photos"
                    >
                      Фотогалерея
                    </a>
                  </li>
                </ul>
              </li>
              <li
                class="nav-item dropdown"
              >
                <div
                  aria-expanded="false"
                  class="menu_link dropdown-toggle"
                  data-bs-toggle="dropdown"
                  role="button"
                >
                  Обучение
                </div>
                <ul
                  class="dropdown-menu dropdown-menu-right"
                >
                  <li>
                    <a
                      class="dropdown-item"
                      href="/centers"
                    >
                      Симуляционные центры
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item"
                      href="/study"
                    >
                      Учебные материалы
                    </a>
                  </li>
                </ul>
              </li>
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="menu_link"
                  href="/conferencies"
                >
                  Конференции
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="menu_link"
                  href="/forum"
                >
                  Форум
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="menu_link"
                  href="/contacts"
                >
                  Контакты
                </a>
              </li>
            </ul>
            <div
              class="nav-item ms-auto headerAccountIcon"
            >
              <a
                aria-current="page"
                class="nav-link"
                href="/login"
              >
                <svg
                  height="32px"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  width="32px"
                >
                  <path
                    d="M170.666667 512a42.666667 42.666667 0 0 0 42.666666 42.666667h323.84l-98.133333 97.706666a42.666667 42.666667 0 0 0 0 60.586667 42.666667 42.666667 0 0 0 60.586667 0l170.666666-170.666667a42.666667 42.666667 0 0 0 8.96-14.08 42.666667 42.666667 0 0 0 0-32.426666 42.666667 42.666667 0 0 0-8.96-14.08l-170.666666-170.666667a42.666667 42.666667 0 1 0-60.586667 60.586667l98.133333 97.706666H213.333333a42.666667 42.666667 0 0 0-42.666666 42.666667zM725.333333 85.333333H298.666667a128 128 0 0 0-128 128v128a42.666667 42.666667 0 0 0 85.333333 0V213.333333a42.666667 42.666667 0 0 1 42.666667-42.666666h426.666666a42.666667 42.666667 0 0 1 42.666667 42.666666v597.333334a42.666667 42.666667 0 0 1-42.666667 42.666666H298.666667a42.666667 42.666667 0 0 1-42.666667-42.666666v-128a42.666667 42.666667 0 0 0-85.333333 0v128a128 128 0 0 0 128 128h426.666666a128 128 0 0 0 128-128V213.333333a128 128 0 0 0-128-128z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
        <form
          class="d-none"
          role="search"
        >
          <input
            aria-label="Search"
            class="me-2 headerSearchInput"
            placeholder="Поиск"
            type="search"
          />
          <button
            class="btn headerSearchBtn"
            type="submit"
          >
            <img
              alt="search"
              src=""
            />
          </button>
        </form>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
